### PR TITLE
Tools: filesize comparison on PR

### PR DESCRIPTION
**Pull request by goransle**
* Adds Gulp scripts that comments on pull requests with a comparison of compiled product sizes before and after changes (see example in comments). 
* Moved a some GitHub query functions from `tools/gulptasks/update-pr-testresults.js` to a `tools/gulptasks/lib/github.js` module.

### CODE REVIEW

File: tools/write-filesizes.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+const mkdirp = require('mkdirp');
+const { join } = require('path');
+const yargs = require('yargs').argv;
+const { getFileSizes } = require('./compareFilesize');
+const log = require('./gulptasks/lib/log');
+
+const files = ['highcharts.src.js'];
```
Comment (TorsteinHonsi) [inline tools/write-filesizes.js:7]: This is the source files with comments and all. We want to inspect the compiled file, `highcharts.js`.

Comment (TorsteinHonsi) [reply tools/write-filesizes.js:7]: Inspecting the `getFileSizes` function, I realize this is built in..

**Comment by goransle** [comment]
Nice!

**Comment by goransle** [comment]
##File size comparisons 
| | master | candidate | difference |
|-------------|-------------|-------------|-------------|
| highcharts | 259133 | 259134 | 1 |
| highcharts, gzipped | 90697 | 90698 | 1 |

**Comment by pawelfus** [comment]
Any chance to add this to other products? Stock, Maps and Gantt? No need for modules, I think.

**Comment by goransle** [comment]
### File size comparison table
| | master | candidate | difference |
|-------------|-------------|-------------|-------------|
| highcharts | 259133 | 259134 | 1 |
| highcharts, gzipped | 90697 | 90698 | 1 |

**Comment by goransle** [comment]
### File size comparison table
| | master | candidate | difference |
|-------------|-------------|-------------|-------------|
| highcharts | 259133 | 259134 | 1 |
| highcharts, gzipped | 90697 | 90698 | 1 |

**Comment by goransle** [comment]
> Nice work!
> 
> Some change requests:
> 
> * Make sure errors pass.
> * Can we have the file sizes in kB so it takes less time for the brain to process? Note that 1 kB is 1024 bytes, so 90697 bytes = 88.6 kB. The difference can be rendered in bytes though.
> * As Pawel suggested, add stats for `highmaps.js` and `highcharts-gantt.js` too.

Will do. The current errors comes from the new scripts not being on the master branch. I think I'll put the CI changes on a separate PR, which will allow us a proper test-run.

**Comment by TorsteinHonsi** [comment]
The kB numbers can be rounded to 1 decimal place.

> I think I'll put the CI changes on a separate PR, which will allow us a proper test-run.

Yes, let's do that.

**Pull request merged**